### ISSUE TITLE
Allow speaking Dictation with the Sphinx backend again

### ISSUE THREAD

**Issue by Danesprite** [dictation-toolbox/dragonfly#69]
The CMU Pocket Sphinx engine implementation does not currently allow matching `Dictation` elements by speaking, although they can be matched with mimic.

The old way of doing this was removed in #62 (release 0.12.0) because it really didn't work well. I'd like to add two ways of using dictation:

1. Speaking Dictation words after utterance breaks via the default language model search (the old behaviour).
2. Allow using a single language model search for matching dictation and grammar words with no utterance breaks.

I think I can re-implement option 1 in a cleaner, more efficient way now. Any parts of grammar rules using `Dictation` elements would not be repeatable. This would be the default behaviour. It would not be difficult to add an engine configuration option to choose either option.

Option 2 would require a custom language model, similar to what Silvius uses. The engine would need to differentiate between dictation and grammar words. This can be done by adjusting the probability score for grammar words in the language model. Grammar reloading and list updates wouldn't work without having some way to reload the model.

@dwks I am guessing this is possible if a custom ngram model file is set in the decoder configuration. It wouldn't be as accurate as Silvius, but I think it would be nice to have this option available! :-)

### TODO

- [ ] Fix a few problems with the pyjsgf DictationGrammar class and release a new version.
- [ ] Re-implement Dictation element support with utterance breaks using a simpler, timer-based approach.
- [ ] Implement Dictation element support relying on the default language model search for both dictation and grammar rule recognitions **without** utterance pauses.
- [ ] Add engine configuration option(s) for dictation support.
- [ ] Adjust (and simplify) dictation tests in test_engine_sphinx_dictation.py appropriately.

**Comment by Danesprite** [comment]
I will only be making dictation work as far as PR #236 makes it work, which is for single dictation utterances using the language model hypothesis if no other grammar rules match. This means that I will **not** be implementing mixed grammar and dictation rules for the Pocket Sphinx engine (e.g. `"type <dictation>"`). My reasoning for this decision is that dictation will work well enough for what Pocket Sphinx is good for (small grammars) and every other SR engine backend doesn't have this limitation.

Once #236 is merged, the following `MappingRule` example and other equivalent rules **will** work if Pocket Sphinx doesn't match what was said to a grammar rule without dictation:

```Python
"<dictation>": Text("%(dictation)s",

```

I will update the Pocket Sphinx engine's documentation page to reflect my decision, what will work in the next version and what (still) won't work.
